trestle/core/commands/author/profile.py

```diff
@@ -536,3 +539,201 @@ def resolve_profile(
+        self.add_argument('-s', '--ssp', help=ssp_help_str, required=True, type=str)
+        profile_help_str = 'Name of the parent profile model in the trestle workspace'
+        self.add_argument('-p', '--profile', help=profile_help_str, required=True, type=str)
+        output_help_str = 'Name of the output generated json Profile'
+        self.add_argument('-o', '--output', help=output_help_str, required=True, type=str)
+        self.add_argument('-vn', '--version', help=const.HELP_VERSION, required=False, type=str)
+
+    def _run(self, args: argparse.Namespace) -> int:
+        try:
+            log.set_log_level_from_args(args)
+            trestle_root: pathlib.Path = args.trestle_root
+
+            if args.profile:
+                if args.profile == args.output:
+                    logger.warning(f'Output profile {args.output} cannot equal parent')
+                    return CmdReturnCodes.INCORRECT_ARGS.value
+
+            return self.initialize_profile(
+                trestle_root=trestle_root,
+                parent_prof_name=args.profile,
+                output_prof_name=args.output,
+                leveraged_ssp_name=args.ssp,
+                version=args.version
+            )
+        except Exception as e:  # pragma: no cover
+            return handle_generic_command_exception(e, logger, 'Profile generation failed')
+
+    @staticmethod
+    def _is_inherited(all_comps: List[ssp.ByComponent]) -> bool:
+        # Fail fast by checking for any non-compliant components.
+        # Must contain provided export statements, no responsibility
+        # statements, and be implemented.
+        for comp in all_comps:
+            if comp.export is None:
+                return False
+
+            if comp.export.responsibilities is not None:
+                return False
+
+            if comp.export.provided is None:
+                return False
+
+            if comp.implementation_status.state != const.STATUS_IMPLEMENTED:
+                return False
+
+        return True
+
+    @staticmethod
+    def update_profile_import(
+        orig_prof_import: prof.Import, leveraged_ssp: ssp.SystemSecurityPlan, catalog_api: CatalogAPI
+    ) -> None:
+        """Add controls to different sections of a profile import based on catalog and leveraged SSP.
+
+        Args:
+            orig_prof_import: The original profile import that will have the control selection updated.
+            leveraged_ssp: SSP input for control filtering
+            catalog_api: Catalog API with access to controls that need to be filtered
+
+        Returns:
+            None
+        """
+        exclude_with_ids: Set[prof.withId] = set()
+        components_by_id: Dict(str, List[ssp.ByComponent]) = {}
+
+        # Create dictionary containing all by-components by control for faster searching
+        for implemented_requirement in leveraged_ssp.control_implementation.implemented_requirements:
+            components: List[ssp.ByComponent] = []
```

Comment:
Pls change variable name to by_components to be more meaningful. I agree that in the end are components, but for readability purposes I´d prefer to call this variable as it is